### PR TITLE
fix: set auction artwork initial sort from querystring

### PR DESCRIPTION
**Pull request by anandaroop**
The type of this PR is: **Fix**

This PR solves [FX-4744]

### Description

We found that auction artwork grids were not respecting an incoming `sort` param in the URL querystring, and instead always sorting by lot number (`sort: sale_position`) —

```
https://www.artsy.net/auction/artsy-auction-street-art-2?sort=-prices
                                                         ↑↑↑↑↑↑↑↑↑↑↑↑ ignored
```

This is because the default value of `sale_position` for the `sort` param was being set later in the request lifecycle, and was thus clobbering the value coming in from the querystring.

This PR hoists that default out of the later location (`AuctionArtworkFilter.tsx`) up into an earlier location (`auctionRoutes.tsx`) so that it can be correctly overriden by the URL param.

The grid is now sorted correctly…

✓ when there is no `sort` param in the URL _(defaults to `sort: sale_position`)_
✓ when there is a `sort` param in the URL _(the specified `sort` is used during the server render)_
✓ when a sort is chosen from the sort options menu  _(the specified `sort` is used during the client refetch)_

[FX-4744]: https://artsyproduct.atlassian.net/browse/FX-4744?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Auction/auctionRoutes.tsx [hunk 1]
```diff
@@ -72,20 +72,31 @@ export const auctionRoutes: AppRouteConfig[] = [
...
+      )
+
+      const variables = {
         slug: params.slug,
         input: {
-          ...initialFilterState,
-          ...getArtworkFilterInputArgs(props.context.user),
+          ...auctionFilterDefaults,
+          ...initialFilterStateFromUrl,
+          ...userSpecificFilterState,
```
Comment (anandaroop) [inline src/Apps/Auction/auctionRoutes.tsx:92]: This now represents the correct order of preference for the filter params.

File: src/Apps/Auction/auctionRoutes.tsx [hunk 2]
```diff
@@ -72,20 +72,31 @@ export const auctionRoutes: AppRouteConfig[] = [
       }
     `,
     prepareVariables: (params, props) => {
-      const initialFilterState = getInitialFilterState(
+      const auctionFilterDefaults = {
+        sort: "sale_position",
+      }
```
Comment (dimatretyak) [inline src/Apps/Auction/auctionRoutes.tsx:77]: nice

**Comment by dimatretyak** [comment]
noticed strange behavior when the user is **not** logged in. trying to understand the cause of the problem


https://user-images.githubusercontent.com/3513494/235003683-e10a81af-dab9-435c-b464-7a5aed37f445.mp4

**Comment by dimatretyak** [comment]
ok, I checked on the [review app](https://auction-sort.artsy.net/) and everything works without problems. It looks like I'm having some problems with the local cache

https://user-images.githubusercontent.com/3513494/235006338-159bd9f6-4cc6-4d03-9c46-33c444b2ba0e.mp4

**Review by dimatretyak — APPROVED** [review]

**Comment by anandaroop** [comment]
Thanks for verifying with the review app!

**Pull request merged**